lib/libc/Kconfig

```diff
@@ -95,7 +94,6 @@ config PICOLIBC
 config NEWLIB_LIBC
 	bool "Newlib C library"
 	select COMMON_LIBC_ABORT
-	depends on !NATIVE_APPLICATION
```

Comment:
Yeah, if you remove the condition here, you should add it to `NEWLIB_LIBC_SUPPORTED`.

Reply:
Changed.
Though note we have `depends on "$(TOOLCHAIN_HAS_NEWLIB)" = "y"` which effectively excludes all builds with host tooling.